### PR TITLE
Fix/custom containers

### PR DESCRIPTION
**Pull request by ericawright**
Status: **Ready for Review** 
Owner: @ericawright
Reviewers: @mikenikles @marlowpayne @jansepar @fractaltheory @stewartyu 
## Changes
- user was unable to add custom classes for appendTo and container options (unless they also used pikabu's built in classes)
- $viewport was not being defined when custom classes used
- $animators array was missing custom class when .pikabu_container class was not there
- custom test fixture built and used for custom appendTo and container
## Issues:

https://github.com/mobify/pikabu/issues/60
## Todos:
- [ ] Engineer +1
## Notes:
- sorry for the spam y'all. This time, all the tests pass, and have made a custom fixture for custom container and appendTo. The tests were incorrectly passing previously because those custom pieces were using defaults instead.
- also, I was incorrectly using .push for a Jquery array instead of .add
## How to Test
- in `pikabu/examples/assets/js/main`, add some custom options such as:
  `container: '.m-pikabu-container',
          appendTo: '.m-pikabu-viewport',`
  to the pikabu declarations (make sure to do this to all 4, or else it will look for the default elements which we will remove)
- in `pikabu/examples/index.html` change line 18 `<div class="pikabu">` remove "pikabu" and add custom class name: ".m-pikabu-viewport"
- also in `pikabu/examples/index.html` change line 25 `<div class="pikabu__container">` remove "pikabu_container" and replace with custom class "m-pikabu-container"
- run `grunt serve`
- go to http://localhost:3000/examples/

### CODE REVIEW

**Comment by fractaltheory** [comment]
Lint error:

screen shot 2016-03-04 at 4 10 55 pm

**Comment by ericawright** [comment]
@fractaltheory lint error fixed

File: dist/pikabu.js [hunk 1]
```diff
@@ -246,7 +247,7 @@
 
             this.$container = this.$pikabu.data('lockup').$container.addClass(classes.CONTAINER);
 
-            this.$pikabu.appendTo(this.options.appendTo ? $(this.options.appendTo) : this.$container);
+            this.$pikabu.appendTo(this.$viewport);
```
Comment (MikeKlemarewski) [inline dist/pikabu.js:246]: I think we may want to keep the previous logic in there. If the user doesn't tell pikabu to `appendTo` anything, then it should probably append to the container. Unless we're planning on changing this behaviour, then we should update the docs.

**Comment by MikeKlemarewski** [comment]
pikabu

I tried testing out these changes, and it seemed to break the left hand menu for me :\

File: dist/pikabu.js (hunk 1 above)
Comment (ericawright) [inline dist/pikabu.js:246]: @MikeKlemarewski  yes, it turns out that when the user does not enter an option, `this.options.appendTo` becomes $('.pikabu') essentially there is always a `this.options.appendTo` and it is === $viewport

Comment (MikeKlemarewski) [inline dist/pikabu.js:246]: This isn't quite the same behaviour as before though. Our docs read:

```
Specify the element Pikabu will be appended to. By default Pikabu will be appended to the lockup container.
```

With this new logic pikabu uses the class `.pikabu__container` for the lockup container. If a user doesn't pass in the `appendTo` option, then the `$viewport` variable will be looking for $('.pikabu') which may be a different element than the lockup container. I'd say we either update the docs to reflect this new behaviour, or revert this one line.

Comment (MikeKlemarewski) [inline dist/pikabu.js:246]: Now I feel crazy because I can't find the version of the docs where I copy pasted that from ^^^

Comment (MikeKlemarewski) [inline dist/pikabu.js:246]: Ahhhhh, I was looking at the docs on the master branch, which are a bit older than the ones on develop.

File: dist/pikabu.js [hunk 2]
```diff
@@ -227,7 +227,7 @@
                     height: this.options.coverage
                 })
                 .lockup({
-                    container: this.options.container,
+                    container: $('.' + classes.CONTAINER),
```
Comment (MikeKlemarewski) [inline dist/pikabu.js:230]: I think this could be left as `this.options.container`. Correct me if I'm mistaken.

Comment (ericawright) [inline dist/pikabu.js:230]: it cannot, now, this.options.container is an empty array when no custom container is set.

**Comment by MikeKlemarewski** [comment]
Looks good to me :+1:

**Pull request merged**